src/components/views/DownloadModModal.vue

```
@@ -60,49 +59,48 @@ import ProfileModList from '../../r2mm/mods/ProfileModList';
                                 try {
-                                    await DownloadModModal.installModAfterDownload(profile, combo.getMod(), combo.getVersion());
+                                    DownloadMixin.downloadProgressCallback(store, assignId, progress, modName, status, err);
                                 } catch (e) {
-                                    return reject(
-                                        R2Error.fromThrownValue(e, `Failed to install mod [${combo.getMod().getFullName()}]`)
-                                    );
+                                    reject(e);
                                 }
                             }
-                            const modList = await ProfileModList.getModList(profile.asImmutableProfile());
-                            if (!(modList instanceof R2Error)) {
-                                const err = await ConflictManagementProvider.instance.resolveConflicts(modList, profile.asImmutableProfile());
-                                if (err instanceof R2Error) {
-                                    return reject(err);
-                                }
+                        );
+                    } catch (e) {
+                        return reject(e);
+                    }
+                    await ProfileModList.requestLock(async () => {
+                        for (const combo of downloadedMods) {
+                            try {
+                                await DownloadModModal.installModAfterDownload(profile, combo.getMod(), combo.getVersion());
+                            } catch (e) {
+                                return reject(
+                                    R2Error.fromThrownValue(e, `Failed to install mod [${combo.getMod().getFullName()}]`)
+                                );
+                            }
+                        }
+                        const modList = await ProfileModList.getModList(profile.asImmutableProfile());
```

As the following issues existed before this PR I think they're out of scope but please take a note that we should check if at least some of the can **and** should be addressed:

- Any error from `.getModList()` is silently ignored
- Depending on how `.resolveConflicts()` is used elsewhere, and how it's implemented internally it could make sense to not pass `modList` as an argument but read it internally (it seems only profile is required to do that). However, keeping `ProfileModList` and  the implemenation of `ConflictManagementProvider` loosely coupled might be the correct idea too
-  `.resolveConflicts()` is called separately for each installed mod in this loop, and it might be unnecessary if calling it only once in the end with the final `modList` yields the same results. However, if we want to fail early to keep the profile from corrupting somehow, the current approach might be the correct idea

---

I'll fix these in the in https://github.com/ebkr/r2modmanPlus/pull/1625

---

For future reference, all of these were in fact not fixed in #1625.